src/twisted/trial/_dist/worker.py

```diff
@@ -284,8 +284,9 @@ def connectionMade(self):
             os.makedirs(self._logDirectory)
         self._outLog = open(os.path.join(self._logDirectory, 'out.log'), 'wb')
         self._errLog = open(os.path.join(self._logDirectory, 'err.log'), 'wb')
-        testLog = open(os.path.join(self._logDirectory, self._logFile), 'wb')
-        self._ampProtocol.setTestStream(testLog)
+        self._testLog = open(
```

Comment:
Both this line, and the one closing the file, are suspiciously similar to the line above -- and by now, there are three copies of those.

It might be worthwhile to consider if this can be abstracted out.